[T09-2] Anakin

thanh-ntt opened the pull request:
@leeyjjoel @tomforge @yujiatay @Lessthanfree

novellius commented on README.adoc:
```diff
@@ -1,28 +1,33 @@
+
+= Anakin
+
+image:https://api.codacy.com/project/badge/Grade/51c8e6629cba4d068c77e39bf0094e4c[link="https://app.codacy.com/app/leeyjjoel/main?utm_source=github.com&utm_medium=referral&utm_content=CS2103-AY1819S1-T09-2/main&utm_campaign=Badge_Grade_Dashboard"]
```
Why are there 2 codacy badges in the ReadMe.md?

novellius commented on build.gradle:
```diff
@@ -207,8 +207,8 @@ asciidoctor {
-        'site-name': 'AddressBook-Level4',
-        'site-githuburl': 'https://github.com/se-edu/addressbook-level4',
+        'site-name': 'Anakin',
+        'site-githuburl': 'https://github.com/CS2103-AY1819S1-T09-2/Main.git',
```
Is the site-githuburl still /Main.git instead of /main.git? Should it contain .git?

yujiatay replied:
Rectified in 68fbd75fd93773293e8e72a2fffcc73bbb961c2f

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -12,7 +12,7 @@ ifdef::env-github[]
 :warning-caption: :warning:
 endif::[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103-AY1819S1-T09-2/Main.git
```
repoURL should not have .git at the back.

yujiatay replied:
Rectified in ccaca8fec2d75e6dcbc61a02aa631e2d7dc1f0cf

novellius commented on docs/UserGuide.adoc:
```diff
@@ -54,207 +54,161 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-* *History* : `history`
-* *Undo* : `undo`
-* *Redo* : `redo`
+* See above please :)
```
Is there a reason to not have a command summary? Perhaps you can add it back once the user guide is filled with more commands.

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -12,7 +12,7 @@ ifdef::env-github[]
+:repoURL: https://github.com/CS2103-AY1819S1-T09-2/Main.git
 
 By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
 
```
Where are the updated user stories, use cases, NFRs and product scope?

yujiatay replied:
Please refer to them at https://github.com/CS2103-AY1819S1-T09-2/main/blob/master/docs/DeveloperGuide.adoc

novellius commented on docs/UserGuide.adoc:
```diff
@@ -54,207 +54,161 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-* Existing values will be updated to the input values.
-* When editing tags, the existing tags of the person will be removed i.e adding of tags is not cumulative.
-* You can remove all the person's tags by typing `t/` without specifying any tags after it.
+*Name will be taken as a string.
```
You need to check whether you have applied the markup syntax correctly.

yujiatay replied:
Rectified in 2e3bcec9b1899c1640c038941df3a79982b1a13e

novellius commented on docs/UserGuide.adoc:
```diff
@@ -54,207 +54,161 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 ****
 
 Examples:
+* `newdeck n/My First Deck`
```
It would be good practice to ensure that your examples have a consistent format across the commands.

novellius commented on src/main/java/seedu/address/model/Anakin_deck/Anakin_Card.java:
```diff
@@ -0,0 +1,65 @@
+/**
+ * Represents a Card inside a Deck.
+ */
+public class Anakin_Card {
```
Class names should not have underscores. Ensure that they follow PascalCase. Perhaps you can rename it to AnakinCard.

kr0stofa replied:
Resolved in  [#146](CS2103-AY1819S1-T09-2/main/pull/146)

novellius commented on src/main/java/seedu/address/model/Anakin_deck/Anakin_Deck.java:
```diff
@@ -0,0 +1,61 @@
+/**
+ * Represents a Deck inside Anakin.
+ */
+public class Anakin_Deck {
```
Same as above comment, the class name needs to be changed.

kr0stofa replied:
Resolved in [#146](CS2103-AY1819S1-T09-2/main/pull/146)

novellius reviewed (commented):
Kindly read the comments for suggestions. Do NOT close this PR. Please update your Developer Guide's appendix sections as stated in the project milestone requirements. Please read the requirements carefully for every week so that you don't miss the milestone requirements. Otherwise, fine work. Remember to tag all other group members (using the @ notation) in the PR description so that they can receive notifications regarding this PR.

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -67,895 +67,250 @@ This project follows https://github.com/oss-generic/process/blob/master/docs/Cod
-.List of per-file attributes, excluding Asciidoctor's built-in attributes
-|===
-|Attribute name |Description |Default value
+*Value proposition*: manage flashcards faster than a typical mouse/GUI driven app
```
Perhaps your value proposition can be more detailed so that users are more tempted to try the app? (What is the benefit of managing flashcards faster than manual/mouse/GUI driven app?)

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -67,895 +67,250 @@ This project follows https://github.com/oss-generic/process/blob/master/docs/Cod
 
-The files in link:{repoURL}/docs/templates[`docs/templates`] controls the rendering of `.adoc` files into HTML5.
-These template files are written in a mixture of https://www.ruby-lang.org[Ruby] and http://slim-lang.com[Slim].
+|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
```
Is Anakin keeping a list of persons in addition to flashcards?

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -67,895 +67,250 @@ This project follows https://github.com/oss-generic/process/blob/master/docs/Cod
 
-[discrete]
-==== `Logic` component
+*Alternate Use Case*
```
Does this mean there are alternative methods to delete a card from the deck? Perhaps you can place them under extensions [E.g. 3b. User selects a deck. (Rest of alternate path)]

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -67,895 +67,250 @@ This project follows https://github.com/oss-generic/process/blob/master/docs/Cod
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `9` or higher installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+. Should be usable on a laptop with average hardware
+. Internal state should be persistent
```
Perhaps the term internal state can be clearer? Does it refer to storage? Or whenever the App is launched, it resumes the previous session state the user has left it? You can define the meaning under glossary.

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -67,895 +67,250 @@ This project follows https://github.com/oss-generic/process/blob/master/docs/Cod
-
-[[Docs-PerFileDocSettings]]
-=== Per-file Documentation Settings
+[appendix]
```
Is there a reason for removing the design, implementation, documentation, testing and devops? These are required in the end product. Remember that the developer guide is supposed to guide new developers in performing certain tasks like testing, setting up documentation etc.

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 Format: `help`
 
-=== Adding a person: `add`
+=== Lists all decks (or all cards) : `list`
```
For command features, in headers like this, remember to mark it as "Coming in V2.0" if its not implemented at the current milestone.

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 
-Deletes the specified person from the address book. +
-Format: `delete INDEX`
+Deletes the specified person from Anakin. +
```
So is Anakin keeping track of persons or decks and cards?

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+`select 2` +
+Enter the 2nd deck in the deck list
+
+=== Navigating out of the current deck : `cd`
```
Perhaps this can be merged with the above [Navigating into a deck] section? They have the same keywords, you can just add a description to specify how to exit from a deck.

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+
+=== Adding a card : `newcard`
+
+Edits an existing card in the specified deck. +
```
Why does it have the same description as editcard?

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+Format: `newcard d/INDEX_OF_DECK [q/QUESTION] [a/ANSWER]`
+
+If user is inside a deck, can omit: d/INDEX_OF_DECK. +
+Format: `newcard [q/QUESTION] [a/ANSWER]`
```
So a card can have no question but an answer, or have a question with no answer?

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+Edits the answer of the first card in the first deck to be `A`.
 
-=== Listing entered commands : `history`
+=== Sort current list by name : `sortname`
```
The term `list` can confuse readers in understanding the action. It can be interpreted as sorting the list of decks in alphabetical order when the actual command can actually sort the cards in a deck. The description and header can be made clearer.

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
+* Only the name is searched.
+* Only full words will be matched e.g. `Han` will not match `Hans`
+* Decks matching at least one keyword will be returned (i.e. `OR` search). e.g. `Hans Bo` will return `Hans Gruber`, `Bo Yang`
```
Feels like the names of the deck resembles people in here while the names used in examples given in this section feel more appropriate.

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-`redo` (reapplies the `clear` command) +
-// end::undoredo[]
+****
+* If FILEPATH is left empty, the file will be created in the same directory as Anakin.
```
What happens if given filepath is invalid?

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-Exits the program. +
-Format: `exit`
+****
+* Note: User must be inside a deck to perform this command.
```
The "Note:" here feels odd when the statement is placed in a box.

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
-* *History* : `history`
-* *Undo* : `undo`
-* *Redo* : `redo`
+A convenient cheat sheet of commands
```
In your command summary, the description of certain commands seems different from that in the earlier section.

novellius commented on README.adoc:
```diff
@@ -1,28 +1,32 @@
+
 ifdef::env-github[]
-image::docs/images/Ui.png[width="600"]
+image:docs/images/ui-mockups/Gameplay-Question.png[width="400"]
```
UI mockups should be 1 screenshot

novellius commented on README.adoc:
```diff
@@ -1,28 +1,32 @@
-
 == Site Map
 
 * <<UserGuide#, User Guide>>
```
You should remove the link to Learning Outcomes in your final product.

novellius commented on src/main/java/seedu/address/logic/Anakinparser/AnakinNewDeckCommandParser.java:
```diff
@@ -0,0 +1,65 @@
+        }
+
+        /*
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
```
Dead code should be removed.

novellius reviewed (commented):
Overall, good job. Most of the documentation is updated. Some points to note:
* ContactUs page needs to be changed (This is your product, they should not be contacting Prof. Damith)
* For features yet to be implemented by the milestone, mark them as "Coming in V2.0" in the UserGuide
* Certain badges does not seem to be following your team repo, but a member's repo's status instead. (E.g. AppVeyor, Codacy)
As usual, do not close the PR. Continue to work on your individual features and upgrade the documentation as needed.

novellius commented on docs/AboutUs.adoc:
```diff
@@ -4,53 +4,67 @@
-image::m133225.jpg[width="150", align="left"]
-{empty}[http://github.com/m133225[github]] [<<johndoe#, portfolio>>]
+=== David Goh
+image::lessthanfree.png[width="150", align="left"]
```
@Lessthanfree Your picture may need to be changed such that your face can be seen more clearly.

novellius commented on README.adoc:
```diff
@@ -1,40 +1,59 @@
 ifdef::env-github[]
-image::docs/images/Ui.png[width="600"]
+image:docs/images/ui-mockups/Gameplay-Question.png[width="400"]
+image:docs/images/ui-mockups/Gameplay-Answer.png[width="400"]
```
You need to have your one of your images named "docs/images/Ui.png" so that our scripts can download it.

novellius commented on docs/UserGuide.adoc:
```diff
@@ -47,214 +52,231 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+|=======================================================================
+|Command | What does it do?
+|`list` | Lists all decks
+|`newdeck [n/NAME]` | Creates a new deck. /n is optional
```
Should it be "n/" instead of "/n"?

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -177,785 +119,440 @@ The `UI` component,
 
-*API* :
-link:{repoURL}/src/main/java/seedu/address/logic/Logic.java[`Logic.java`]
+The `LogicManager` contains an `AnakinModel`, an `AnakinParser`, and a `CommandHistory`.
```
I cant seem to find `AnakinModel` in your diagram. Your diagram has an orange dotted line pointing from `LogicManager` to `CommandHistory`. Does that mean that it contains and modifies `CommandHistory`? [It was not stated in the Legend]

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -177,785 +119,440 @@ The `UI` component,
+image::AnakinModelClassDiagram.png[width="900", align="left"]
 
-*API* : link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model.java`]
+Structure of the Model Component
```
Try to standardize your class diagram appearance.

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -177,785 +119,440 @@ The `UI` component,
-*API* : link:{repoURL}/src/main/java/seedu/address/model/Model.java[`Model.java`]
+Structure of the Model Component
 
 The `Model`,
```
I believe your model is actually called `AnakinModel` instead of `Model`? [The diagram does not reflect the new name]

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -177,785 +119,440 @@ The `UI` component,
+* Anakin#updateCard( Card target, Card editedCard) - Update the information of a card inside the current deck.
 
-// end::dataencryption[]
+These operations are exposed in the Model interface as: Model#addCard(Card card), Model#deleteCard(Card card), Model#updateCard(Card target, Card editedCard) respectively.
```
Remember to rename `Model` to the actual name of the interface.

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -177,785 +119,440 @@ The `UI` component,
+The following sequence diagram shows how the `newcard` operation works:
 
-Certain properties of the application can be controlled (e.g App name, logging level) through the configuration file (default: `config.json`).
+image::NewCard sequence diagram.png[width="790", align="left"]
```
- Make sure the arrows touch the activation bars.
- Would *alt* be more suitable than *opt*? If exceptions are thrown, would the rest of the operation still occur in `Anakin` and `UniqueCardList`? Using *opt* would imply that an Exception instance is created, but add operation still continues.
- It is fine to leave out cases where exception is thrown. You just need to explain the main scenario of `newcard` operation.

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -177,785 +119,440 @@ The `UI` component,
 
-[TIP]
-Attributes left unset in the `build.gradle` file will use their *default value*, if any.
+image::NewDeck.png[width="790", align="left"]
```
The comments for the above Sequence diagram applies here.

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -177,785 +119,440 @@ The `UI` component,
+Step 4. The user continues reviewing the rest of the cards in the deck.
 
-=== Site Template
+Step 5. Card A is shown to the user more regularly when he reviews the same deck in the future.
```
Do try to add more information on the implementation (Diagrams and its explanations can help) (Design considerations can help too)

novellius reviewed (commented):
Some comments:
- You are supposed to tag your latest commit (before the deadline) as v1.2.
- You have to tag features listed in UserGuide as "Coming in v2.0" if it is not implemented yet.
- Do continue to update your DeveloperGuide, and watch out for mistakes in your diagrams.
- Try to standardize your component diagrams.
- Try to fix CheckStyle errors so that Travis can actually run tests to see if your code is working as intended.
Please read the comments, and ask questions if there is anything unclear.
Keep up the good work.

kr0stofa committed:
Update Tests

And Formatting

kr0stofa committed:
Merge pull request #166 from Lessthanfree/ImportExportFeature

Import Export feature

thanh-ntt committed:
Fix bugs related to import deck command

kr0stofa committed:
Update exception handling for ImportDeckCommand

Also update exception messages for EditDeck and NewDeck

kr0stofa committed:
Update exception handling for ImportDeckCommand

Also update exception messages for EditDeck and NewDeck

thanh-ntt committed:
Fix style

Add Javadoc comment

thanh-ntt committed:
Merge pull request #168 from truongthanh2606/master

Fix bugs related to import deck command

kr0stofa committed:
Move some command messages to Messages.java

kr0stofa committed:
Move some command messages to Messages.java

kr0stofa committed:
Formatting

kr0stofa committed:
Formatting

kr0stofa committed:
More Formatting

kr0stofa committed:
More Formatting

J0 committed:
Add FindCommandSystemTest Deck level checks (#165)

* Add FindCommandSystemTest Deck level checks

* Fix Style Errors

* Fix Style Errors

* Add Deck_G and Deck_H

* Fix Style Errors

* Change person to deck

* Change AddressbookMessages to Messages

Co-Authored-By: leeyjjoel <lee.yi.jie.joel@gmail.com>

kr0stofa committed:
Improve tests

Added parser test and more test for commands

kr0stofa committed:
Merge branch 'LocalFixImportExceptions' into FixImportExceptions

kr0stofa committed:
Formatting

thanh-ntt committed:
Resolve issue #162, update exceptions thrown with cards operation

Card-level operations should throw DeckNotFoundException when they are execute outside a deck
Issue #162 is fixed with a temporary solution

thanh-ntt committed:
Fix style

kr0stofa committed:
Fix ExportDeckCommandTest

kr0stofa committed:
Remove 2 tests from ExportDeckCommandTest

kr0stofa committed:
More Formatting

kr0stofa committed:
Merge pull request #170 from Lessthanfree/FixImportExceptions

Fix import exceptions

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Fix style

thanh-ntt committed:
Add list command and tests

Fix import bug (when import file name string has less than 4 characters)

thanh-ntt committed:
Merge pull request #171 from truongthanh2606/master

Resolve issue #162, add list command, update exceptions thrown with cards operation

yujiatay committed:
Add ability to review decks (#178)

* Remove todo from editcard command

* Add list command to parser

* Abstract deck and card lists into EditView

* Add play command, parser and start gameplay event

* Add end command for deck review and refactor play to review

* Refactor end to endreview command

* Add flipcard command

* Add prevcard and nextcard commands

* Refactor current card pointer and deck review state to model

* More refactoring

* Throw exceptions when trying to execute non-review methods during review

* Resolve merge conflicts

yujiatay committed:
Uncomment system tests

yujiatay committed:
Fix styles

yujiatay committed:
Fix styles

yujiatay committed:
Fix help command system test

kr0stofa committed:
Merge pull request #179 from yujiatay/fix-systemtests

Fix system tests

kr0stofa committed:
Update user guide (#180)

* Update UserGuide and build.gradle

* Update Ui.png

Rename old image to oldUi.png

* Remove AB4 team pictures

* Update UG with review commands

yujiatay committed:
Clear nodes when updating with new card (#182)

yujiatay committed:
Change relative paths for readme images

yujiatay committed:
Remove site-seedu

yujiatay committed:
Add sections 4 to 6 to dev guide

yujiatay committed:
Spacing magic for user guide

thanh-ntt committed:
Merge pull request #183 from yujiatay/update-user-guide

Misc fixes for docs

thanh-ntt committed:
Add project portfolio

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

kr0stofa committed:
Devguide Update

yujiatay committed:
Update docs (#184)

* Add review commands to command summary

* Formatting user guide

* Remove gitter badge

* More formatting

* Update description for card level commands

* Update description for deck review commands

* Update command summary

* Add v1.4 and more reformatting

* Update export command

* Update classify command

kr0stofa committed:
Merge branch 'master' into updateDevGuide

tomforge committed:
Performance tracking - Add classify command and related utils (#186)

* update card model for performance tracking

* added classify card command

* enhance performance enum for convenience

* add performance parsing

* enable CRUD while reviewing deck to update performance

* rephrase classification success message

* update userguide and docs

* style and comments

* update command summary

kr0stofa committed:
Update DevGuide

Reflect new feature of `classify`

kr0stofa committed:
Merge branch 'master' into updateDevGuide

kr0stofa committed:
Merge pull request #188 from Lessthanfree/updateDevGuide

Devguide Update

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Add tests for 4 commands: review, endreview, nextcard, previouscard

Add tests for 4 commands and delete SelectCardCommand

thanh-ntt committed:
Fix style

thanh-ntt committed:
Update project portfolio

thanh-ntt committed:
Modify tests for endreview, previouscard and nextcard commands

thanh-ntt committed:
Merge pull request #192 from truongthanh2606/master

 Add tests for 4 commands, add project portfolio

thanh-ntt committed:
Modify card and deck-level commands, update userguide, developerguide

Block card-level operations while user is not inside any decks
Block deck-level operations while user is inside a deck
Update project portfolio
Update user guide, developer guide

thanh-ntt committed:
Fix test classes

kr0stofa committed:
Attempt to fix undo-cd bug

Create a new method in Anakin to swapCards which creates a new UniqueCardList instead of setting each Card, which overwrites the card list.

kr0stofa committed:
Attempt to fix undo-cd bug

Create a new method in Anakin to swapCards which creates a new UniqueCardList instead of setting each Card, which overwrites the card list.

kr0stofa committed:
Implement a currentDeck in Anakin

kr0stofa committed:
Implement a currentDeck in Anakin

thanh-ntt committed:
Merge pull request #195 from truongthanh2606/master

 Modify card and deck-level commands, update userguide, developerguide

thanh-ntt committed:
Update project portfolio

yujiatay committed:
Update deck review (#189)

* Disable relevant commands when in review mode

* Format find command message_usage

* Format message_usage for commands

* Update unit tests

* Update version number

* Update default GUI window dimensions

* Update MESSAGE_CURRENTLY_REVIEWING_DECK

Co-Authored-By: yujiatay <25505260+yujiatay@users.noreply.github.com>

* Update version number to v1.3.1

* Fix checkstyle

* Block undo/redo while in review mode

* Fix duplicate classify command on user guide

* Update with ppp

tomforge committed:
update proj portfolio (#196)

thanh-ntt committed:
Update developer guide, project portfolio

thanh-ntt committed:
Merge branch 'master' into master

thanh-ntt committed:
Update project portfolio

kr0stofa committed:
Implement justEnteredDeck

kr0stofa committed:
Implement justEnteredDeck

thanh-ntt committed:
Update developer guide, project portfolio (#199)

* Update project portfolio

* Update developer guide, project portfolio

* Update project portfolio

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

kr0stofa committed:
Add justEnteredDeck flag to block undo

Blocks undo for the cd command

kr0stofa committed:
Add justEnteredDeck flag to block undo

Blocks undo for the cd command

kr0stofa committed:
Fix import exception handling and add portfolio

Also formatting

kr0stofa committed:
Fix import exception handling and add portfolio

Also formatting

yujiatay committed:
Update ui component diagram

kr0stofa committed:
Merge branch 'master' into cdUnredoable

kr0stofa committed:
Merge pull request #200 from yujiatay/update-ui-diagram

Update ui component diagram

kr0stofa committed:
Merge branch 'master' into cdUnredoable

kr0stofa committed:
Refactor short method name and fix typos

kr0stofa committed:
Combine similar model components

Join isInsideDeck with currentDeck

kr0stofa committed:
Merge branch 'cdUnredoablelocal' into cdUnredoable

kr0stofa committed:
Format UG,DG and portfolio

kr0stofa committed:
Clean up, fix broken tests

kr0stofa committed:
Merge pull request #201 from Lessthanfree/cdUnredoable

Fix undo-cd bug

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

J0 committed:
Update user guide, add leeyjjoel.adoc (#202)

* Add blurb about Anakin

* Add documentation for showDue command

* Add team document

* Update Highlights in userdocs

* Update name

Co-Authored-By: leeyjjoel <lee.yi.jie.joel@gmail.com>

* Update command name

Co-Authored-By: leeyjjoel <lee.yi.jie.joel@gmail.com>

kr0stofa committed:
Initial commit

thanh-ntt committed:
Change the logic of executing commands

Command exception is now checked in Logic component (in the Command class itself) instead of being checked in Model component.
This reduce the possible bugs but increase coupling between Logic and Model component

thanh-ntt committed:
Fix style

thanh-ntt committed:
Fix style

thanh-ntt committed:
Merge pull request #209 from truongthanh2606/master

Change the logic of executing commands

kr0stofa committed:
Add tests to Parser Test and ReviewCommandParser tests

thanh-ntt committed:
Resolve issues #211, #212, #213, #214, #217

Update User Guide
Update project portfolio

thanh-ntt committed:
Resolve issues #211, #212, #213, #214, #217 (#218)

Update User Guide
Update project portfolio

Commit:
Add LogicManagerTest

Commit:
Change exportdeck to export in user guide

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Merge pull request #238 from leeyjjoel/LogicManagerTest

Add LogicManagerTest

kr0stofa committed:
Add ClassifyCard Parser test

J0 committed:
Update userguide on multiple arguments and import format (#233)

* Update Userguide on multiple arguments and import format

* Update Userguide on newcard command behavior

* Add new line in error message for new card

* Update Userguide to accurately reflect commands

* Update Userguide

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Add messages to `undo` command

thanh-ntt committed:
Fix style

thanh-ntt committed:
Make Travis happy

thanh-ntt committed:
Merge pull request #241 from truongthanh2606/master

 Add messages to `undo` command

thanh-ntt committed:
Update message for `redo`, resolve issue #219

Create a deep copy of Anakin for each state instead of a shallow copy.
`cd` and `cd ..` are now undoable, they will commit Anakin.

thanh-ntt committed:
Merge pull request #243 from truongthanh2606/master

Update message for `redo`, resolve issue #219

kr0stofa committed:
Fix Formatting

tomforge committed:
streamline difficulties, use only "easy", "normal", and "hard"

tomforge committed:
remove default difficulty, use "normal" instead

tomforge committed:
elaborate on difficulty and the use of the classify command

tomforge committed:
correct naming for PPP

kr0stofa committed:
Fix Imports

tomforge committed:
use times reviewed instead of times classified

tomforge committed:
fixes #187

tomforge committed:
rename Performance string to type

tomforge committed:
streamline Performance parsing

kr0stofa committed:
Update ClassifyCommandParser test and SortCommandTest

J0 committed:
Add newdeckcommand system tests (#242)

* Add Deckutil

* Fix Style Errors

* Add newcardcommandsystemtest

* Fix Style Errors

* Remove trailing whitespace

* Remove trailing whitespace

* Update contributions for leeyjjoel

* Update undoredo to include last command word

* Change person to deck in editcommand

* Change person to deck in deckutil

* Remove trailing whitespace

* Remove trailing whitespace

tomforge committed:
streamline performance enum and parsing

tomforge committed:
move string to performance parsing into performance enum

will be used outside of logic as well

tomforge committed:
add valid performance check in Performance enum

tomforge committed:
add Card meta fields to storage

tomforge committed:
display difficulty on UI

tomforge committed:
fix styles

tomforge committed:
Merge branch 'master' into classify-enhancements

tomforge committed:
update UG/DG for classify

tomforge committed:
fix styles, add comments

tomforge committed:
add performance to card test

tomforge committed:
use classification to track times reviewed

tomforge committed:
add performance ParserUtil tests

kr0stofa committed:
Merge pull request #245 from tomforge/classify-enhancements

Classify enhancements

kr0stofa committed:
Adapt tests to Classify command changes

Also adapt to Performance changes

kr0stofa committed:
Fix build errors

kr0stofa committed:
Fix Format

kr0stofa committed:
Add Question tests to ParserUtilTest

kr0stofa committed:
Fix import and variable declaration

kr0stofa committed:
Comment out asciidoc errors in johndoe.adoc

thanh-ntt committed:
Update user guide, project portfolio

thanh-ntt committed:
Update user guide, project portfolio (#247)

kr0stofa committed:
Update my docs

kr0stofa committed:
Merge branch 'master' into ParserTest

kr0stofa committed:
Merge pull request #240 from Lessthanfree/ParserTest

Add Parser integration test and Several Parser unit tests

yujiatay committed:
Fixes #208, remove method call of unused browserpanel

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

kr0stofa committed:
Merge pull request #251 from yujiatay/bug-fix

Fixes exception thrown on exit issue #208

kr0stofa committed:
Update PPP

J0 committed:
Add autocomplete functionality (#248)

Add autocomplete function

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

yujiatay committed:
Add more tests to increase coverage (#250)

* Fix commandbox and mainwindowclose tests

* Added typical anakin in deck review mode

* Added Flip Card command test

* Update help command test

* Update flip card command test

* Add classify command test

* Fix checkstyle

* Add deck review card test

* Fix checkstyle

* Remove browser panel

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Add tests

thanh-ntt committed:
Fix style

kr0stofa committed:
Update PortManager test

kr0stofa committed:
Formatting

thanh-ntt committed:
Merge pull request #255 from truongthanh2606/master

Add tests

kr0stofa committed:
Add jadoc comment

kr0stofa committed:
Merge branch 'master' into PortManagerTest

tomforge committed:
add sampleDataUtil tests

tomforge committed:
formatting

kr0stofa committed:
Enhance ExportDeckCommand test

tomforge committed:
fix formatting

tomforge committed:
fix style

kr0stofa committed:
Merge pull request #256 from Lessthanfree/PortManagerTest

Portmanager and ExportDeckCommand unit tests

tomforge committed:
Merge branch 'master' into add-tests

thanh-ntt committed:
Merge pull request #257 from tomforge/add-tests

Add SampleDataUtil tests

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

J0 committed:
Add system tests for edit deck command (#244)

* Update user guide

* Add messages for EditDeckCommand

* Add initial editCommand system tests

* Add test to redo delete command

* Fix Style Errors

* Fix Style Errors

* Fix Style Errors

* Fix Style Errors

* Fix Style Errors

* Update SampleDataTest

* Remove unused import

* Fix import order

* Add additional system test for edit command

* add final editcommand system tests

* Remove unused import

* Remove commented regions

* Remove unused imports

* Fix style errors

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Resolve issue #228 #249

Commit Anakin after the `endreview` command instead of after each `classify`

thanh-ntt committed:
Update user guide and project portfolio

thanh-ntt committed:
Add logging info

thanh-ntt committed:
Resolve issue #228 #249 (#258)

* Update user guide, project portfolio

* Resolve issue #228 #249

Commit Anakin after the `endreview` command instead of after each `classify`

* Update user guide and project portfolio

* Add logging info

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

novellius commented on docs/DeveloperGuide.adoc:
```diff
@@ -227,104 +171,345 @@ image::StorageClassDiagram.png[width="800"]
+The following sequence diagram shows how the `newcard` operation works:
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+image::NewcardSequenceDiagram.png[width="790", align="left"]
```
For Sequence Diagrams, ensure that the arrows touch the activation bars, and they should not end within the bar.

novellius reviewed (commented):
I have reviewed the DeveloperGuide only. Generally the content is okay except some minor issues in your sequence diagrams. Ensure that the classes' names in the DG match your the actual names of the classes.

kr0stofa committed:
Reposense code

Commit:
Add Performance sort command

Commit:
Add review score and next review date to card

Commit:
Change sort to accept sort types

Commit:
Modify card list to perform performance sort

Commit:
Augment performance enum

Commit:
Add documentation for Card methods

Commit:
Fix style errors

kr0stofa committed:
Update documents and add 5 screenshots

thanh-ntt committed:
Merge pull request #263 from Lessthanfree/Documentation

Enhance documents for PPP

Commit:
Acknowledge Jethro Kuan for suggestions

Commit:
Integrate performance sort to parser

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-2/main

Commit:
Add bias, update threshold

Commit:
Refactor if else loops

Commit:
Update isCorrect core calculation in Performance

Commit:
Remove reviewed property of card

Commit:
Change Alphabetical to Lexicographical sort

Commit:
Change Alphabetical to Lexicographical sort

Commit:
Add documentation for Card methods

Commit:
Add initial helper methods for performance sort tests

Commit:
Update PPP, fix style errors

Commit:
Remove unused constant

kr0stofa committed:
Shorten document (#264)

J0 committed:
Merge branch 'master' into master

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

tomforge committed:
Update ug/dg/ppp (#265)

* update ug/dg/ppp (wip)

* add preface to Storage component DG

* edit Performance tracking  section of DG

* complete PPP

kr0stofa committed:
Merge branch 'master' into master

Commit:
Update user guide with rank command

Commit:
Merge branch 'master' of https://github.com/CS2103-AY1819S1-T09-2/main

Commit:
Rename test

thanh-ntt committed:
Update project portfolio, developer guide

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

J0 committed:
Add scheduling for cards (#262)

* Add Performance sort command

* Add review score and next review date to card

* Change sort to accept sort types

* Modify card list to perform performance sort

* Augment performance enum

* Add documentation for Card methods

* Fix style errors

* Acknowledge Jethro Kuan for suggestions

* Integrate performance sort to parser

* Add bias, update threshold

* Refactor if else loops

* Update isCorrect core calculation in Performance

* Remove reviewed property of card

* Change Alphabetical to Lexicographical sort

* Change Alphabetical to Lexicographical sort

* Add documentation for Card methods

* Add initial helper methods for performance sort tests

* Update PPP, fix style errors

* Remove unused constant

thanh-ntt committed:
Update user guide, project portfolio (#266)

* Update user guide, project portfolio

* Resolve issue #228 #249

Commit Anakin after the `endreview` command instead of after each `classify`

* Update user guide and project portfolio

* Add logging info

* Update project portfolio, developer guide

yujiatay committed:
Update PPP (#267)

* Update Developer Guide

* Add review diagrams

* Update user guide

* Update PPP

* update UG

Commit:
Remove unused import

Commit:
Resolve merge conflicts

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Update PPP

kr0stofa committed:
Merge pull request #269 from truongthanh2606/master

 Update PPP

kr0stofa committed:
Fix formatting of import documentation

kr0stofa committed:
Fix formatting of import documentation

tomforge committed:
Add ability to filter by card difficulty with list command (#271)

* add performance tests

* Remove use of dummy Performance

* add card performance matching predicate

* edit list command to take in and filter by performance arguments

* add comments

* fix style

* fix style

* add listcommand tests

* update user guide

J0 committed:
Merge branch 'master' into finalDocs

kr0stofa committed:
Update User Guide

kr0stofa committed:
Add Manual Testing Appendix to DevGuide

kr0stofa committed:
Merge remote-tracking branch 'Lessthanfree/finalDocs' into finalDocs

kr0stofa committed:
Fix typo in Userguide

thanh-ntt committed:
Update Devguide

Co-Authored-By: Lessthanfree <35485082+Lessthanfree@users.noreply.github.com>

thanh-ntt committed:
Fix Typo in Devguide

Co-Authored-By: Lessthanfree <35485082+Lessthanfree@users.noreply.github.com>

thanh-ntt committed:
Update Devguide

Co-Authored-By: Lessthanfree <35485082+Lessthanfree@users.noreply.github.com>

kr0stofa committed:
Tweak Devguide appendix

thanh-ntt committed:
Update ppp

Fix bug: developer guide does not show image content

kr0stofa committed:
Merge pull request #272 from Lessthanfree/finalDocs

Fix UserGuide and Add appendix to Dev Guide

thanh-ntt committed:
Merge branch 'master' into master

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

J0 committed:
[WIP]: Update PPP (#268)

Update PPP

Commit:
Merge branch 'master' of https://github.com/leeyjjoel/Main

Commit:
Resolve merge conflicts

kr0stofa committed:
Merge branch 'master' into master

thanh-ntt committed:
Merge pull request #273 from truongthanh2606/master

Update ppp

thanh-ntt committed:
Merge remote-tracking branch 'origin/master'

thanh-ntt committed:
Fix image not shown bug

New card sequence diagram is now shown in developer guide

Commit:
Fix Grammar add developer guide section

J0 committed:
Merge branch 'master' into updateDevGuide

thanh-ntt committed:
Merge pull request #274 from truongthanh2606/master

 Fix image not shown bug

J0 committed:
Merge branch 'master' into updateDevGuide

Commit:
Fix style errors

Commit:
Fix style errors

thanh-ntt committed:
Merge pull request #275 from leeyjjoel/updateDevGuide

Update Developer Guide and PPP

thanh-ntt committed:
Update Developer guide

thanh-ntt committed:
Merge pull request #277 from truongthanh2606/master

Update Developer guide

J0 committed:
Fix grammatical errors in PPP (#278)

* Fix style errors

* Fix style errors
